src/eval/func.rs

```diff
@@ -290,74 +293,77 @@ impl Closure {
         depth: usize,
         mut args: Args,
     ) -> SourceResult<Value> {
-        let closure = match &this.repr {
-            Repr::Closure(closure) => closure,
-            _ => panic!("`this` must be a closure"),
-        };
-
-        // Don't leak the scopes from the call site. Instead, we use the scope
-        // of captured variables we collected earlier.
-        let mut scopes = Scopes::new(None);
-        scopes.top = closure.captured.clone();
-
-        // Evaluate the body.
-        let vt = Vt { world, tracer, provider, introspector };
-        let mut vm = Vm::new(vt, route, closure.location, scopes);
-        vm.depth = depth;
-
-        // Provide the closure itself for recursive calls.
-        if let Some(name) = &closure.name {
-            vm.define(name.clone(), Value::Func(this.clone()));
-        }
+        stacker::maybe_grow(32 * 1024, 2 * 1024 * 1024, move || {
```

Comment:
i think we could get rid of all this indent here by moving this around actual recursive call below (`closure.body.eval(&mut vm)`)